exporter.go

```diff
@@ -0,0 +1,77 @@
+package mrT
+
+import (
+	"io"
+
+	"github.com/PathDNA/fileutils/shasher"
+	"github.com/itsmontoya/async/file"
+	"github.com/itsmontoya/seeker"
+)
+
+func newExporter(m *MrT, w io.Writer, txnID string) (e exporter) {
+	e.m = m
+	e.w = w
+	e.txnID = txnID
+
+	e.mf = NewMatch(txnID)
+	return
+}
+
+type exporter struct {
+	txnID string
+
+	m  *MrT
+	w  io.Writer
+	hw *shasher.HashWriter
+	mf *Match
+}
+
+func (e *exporter) exportFrom(f file.Interface, s *seeker.Seeker) (err error) {
+	var ltid string
+	if ltid, err = peekLastTxn(s); err == nil && ltid == e.txnID {
```

Comment:
Where do we check for this err not being nil?

Reply:
So if err is not nil, that means we have zero transactions. I could also ensure ltid is not empty, but I figured checking for lack of error was more correct.

Reply:
Got it